core/src/com/unciv/ui/multiplayer/FriendPickerList.kt

```diff
@@ -0,0 +1,56 @@
+package com.unciv.ui.multiplayer
+
+import com.badlogic.gdx.scenes.scene2d.ui.Table
+import com.badlogic.gdx.scenes.scene2d.ui.TextButton
+import com.badlogic.gdx.scenes.scene2d.ui.VerticalGroup
+import com.unciv.logic.multiplayer.FriendList
+import com.unciv.ui.newgamescreen.PlayerPickerTable
+import com.unciv.ui.utils.BaseScreen
+import com.unciv.ui.utils.onClick
+
+class FriendPickerList(
+    playerPicker: PlayerPickerTable,
+    onSelected: (String) -> Unit
+) : VerticalGroup() {
+
+    private val friendDisplays = mutableMapOf<String, FriendDisplay>()
+    private var friendList: MutableList<FriendList.Friend>
+
+    init {
+        padTop(10f)
+        padBottom(10f)
+
+        friendList = playerPicker.getAvailableFriends().toMutableList()
+
+        for (friend in friendList) {
+            addFriend(friend, onSelected)
+        }
+    }
+
+    private fun addFriend(friend: FriendList.Friend, onSelected: (String) -> Unit) {
+        val friendDisplay = FriendDisplay(friend, onSelected)
+        friendDisplays[friend.name] = friendDisplay
+        addActor(friendDisplay)
+        children.sort()
+    }
+}
+
+private class FriendDisplay(
+    friend: FriendList.Friend,
+    private val onSelected: (String) -> Unit
+) : Table(), Comparable<FriendDisplay> {
```

Comment:
It is used in FriendPickerList - Line 34 to sort the friends, but now that I think of it, it would probably be better to keep the order that they're in when viewing the friends list